classes/class-p4-master-site.php

```diff
@@ -674,9 +674,15 @@ public function p4_auto_generate_excerpt( $post_id, $post ) {
 			remove_action( 'save_post', [ $this, 'p4_auto_generate_excerpt' ], 10 );
 
 			// Generate excerpt text.
-			$post_excerpt   = strip_shortcodes( $post->post_content );
-			$post_excerpt   = apply_filters( 'the_content', $post_excerpt );
-			$post_excerpt   = str_replace( ']]>', ']]&gt;', $post_excerpt );
+			preg_match( '<p>(.*?)</p>', $post->post_content, $match_paragraph );
+			if ( count( $match_paragraph ) > 0 ) {
+				$post_excerpt = $match_paragraph[1];
+				$post_excerpt = apply_filters( 'the_content', $post_excerpt );
```

Comment:
You could move this to after the if-statement so we only have to call it once. Not a problem I think if `str_replace( ']]>', ']]&gt;', $post_excerpt )` is also called then for both cases